### PR TITLE
refactor(load_checks_to_execute): Refactor function and add tests

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context

Please include relevant motivation and context for this PR.

### Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/check/check.py [hunk 1]
```diff
@@ -607,22 +613,32 @@ def update_audit_metadata(
...
-    return checks
+    Returns a set of checks from the given services
+    """
+    try:
+        checks = set()
+        service_list = [
+            "awslambda" if service == "lambda" else service for service in service_list
+        ]
+        for service in service_list:
+            modules = recover_checks_from_provider(provider, service)
```
Comment (n4ch04) [inline prowler/lib/check/check.py]: Here `module` is not descriptive, why not `service_checks` or something like that?
It seems that you are handling a different entity

File: prowler/lib/check/check.py [hunk 2]
```diff
@@ -607,22 +613,32 @@ def update_audit_metadata(
...
+            if not modules:
+                logger.error(f"Service '{service}' does not have checks.")
+
+            else:
+                for check_module in modules:
```
Comment (n4ch04) [inline prowler/lib/check/check.py]: Then :
```
for check in service_checks:
...
```

File: prowler/lib/check/checks_loader.py [hunk 3]
```diff
@@ -22,73 +22,99 @@ def load_checks_to_execute(
...
-        if service_list:
-            checks_to_execute = (
-                recover_checks_from_service(service_list, provider) & checks_to_execute
-            )
+    """Generate the list of checks to execute based on the cloud provider and the input arguments given"""
+    try:
+        # Local subsets
+        checks_to_execute = set()
+        check_aliases = {}
+        check_severities = {
```
Comment (n4ch04) [inline prowler/lib/check/checks_loader.py:30]: I know that severities are not gonna change in the short term, but having this  here hardcoded can create a point of failure.
What do you think?

**Review by n4ch04 — CHANGES_REQUESTED** [review]

File: prowler/lib/check/check.py (hunk 1 above)
Comment (jfagoagas) [reply prowler/lib/check/check.py]: Totally agree with you!

File: prowler/lib/check/check.py (hunk 2 above)
Comment (jfagoagas) [reply prowler/lib/check/check.py]: Donete!

File: prowler/lib/check/checks_loader.py (hunk 3 above)
Comment (jfagoagas) [reply prowler/lib/check/checks_loader.py:30]: This is a good point but I'd say to leave it that way for now and came back after some time, what do you think?

Comment (n4ch04) [reply prowler/lib/check/checks_loader.py:30]: Yes, it does not have an easy solution, and since severity is not gonna change for now we can leave it here

**Review by n4ch04 — APPROVED** [review]

**Pull request merged**